### PR TITLE
Mahmoud into master

### PR DESCRIPTION
**Pull request by mhmodo10**
Added training using an existing model (transfer learning)
Added function to resume training.

### CODE REVIEW

File: scripts/Model.py [hunk 1]
```diff
@@ -138,39 +138,94 @@ def schedular(self, epoch_index, lr):
...
+        model.fit(self.Utils.LoadH5File(X_train_path)/255,
+        self.Utils.LoadH5File(y_train_path)/255,
         batch_size = 1,
         epochs= num_of_epochs,
         callbacks=[CustomLearningRateScheduler(self.schedular), early_stop, tensorboard, model_checkpoint_callback])
+    
 
-model = SuperResModel()
-model.PredictAndShowImage(model.loadModel('super_res.h5', compile = False), data_path=r"D:\HBO\MinorAi\Div2kx4\valid_lr", read_from_directory = True)
+    """Resumes training after sudden stop of a model and
```
Comment (Joostdambrink) [inline scripts/Model.py]: Ff je comment afmaken haha

File: scripts/Model.py [hunk 2]
```diff
@@ -138,39 +138,94 @@ def schedular(self, epoch_index, lr):
...
+        batch_size = 1,
+        epochs= num_of_epochs,
+        initial_epoch = latest_epoch,
+        callbacks=[CustomLearningRateScheduler(self.schedular), early_stop, tensorboard, model_checkpoint_callback])
+        
 
-#model.TrainModel( r"D:\HBO\MinorAi\PickleFiles\X_train_1.h5", r"D:\HBO\MinorAi\PickleFiles\y_train_1.h5", num_of_epochs = 1)
\ No newline at end of file
+
+    """Calculates epoch index based on learning rate for
```
Comment (Joostdambrink) [inline scripts/Model.py]: Deze ook denk ik

File: scripts/Model.py [hunk 3]
```diff
@@ -138,39 +138,94 @@ def schedular(self, epoch_index, lr):
...
+
+    Returns:
+        (int) : epoch number
+    """
+    def GetEpochFromLearningRate(self,lr):
+        count = 0
+        start_value = 0.10000
+        lr_rounded = round(lr,5)
+        while start_value > lr_rounded:
+            start_value = start_value / 2
```
Comment (Joostdambrink) [inline scripts/Model.py]: `start_value /= 2`
is more efficient

**Review by Joostdambrink — CHANGES_REQUESTED** [review]
Nice work mahmoud, very impressed

File: scripts/Model.py (hunk 1 above)
Comment (Lorenzokrd) [reply scripts/Model.py]: 😂🤣😂🤣👌

**Review by Joostdambrink — APPROVED** [review]
Much better and much more efficient now GTM!!!

**Pull request merged**